src/hooks/useMutation/index.ts

```diff
@@ -0,0 +1,55 @@
+import { useState } from 'react';
+import { IMutationState, TArgsMutationFunction, TUseMutation } from './useMutation.types';
+import { AxiosError } from 'axios';
+import { api } from 'services';
+
+const useMutation = (): TUseMutation & IMutationState => {
+  const [state, setState] = useState<IMutationState>({
```

Comment:
create seperate states for these
const [isLoading, setIsLoading] = useState(false);
...